### ISSUE TITLE
Implement a case statement

### ISSUE THREAD

**Issue by tristil** [rubysolo/dentaku#60]
I have need of a case statement to handle looking up values. This is the way I would be inclined to implement it (I pass values in from a yaml file):

```
price: 100
sale: 'black friday'
price_after_sale:
  " CASE sale
       WHEN 'black friday':  "price - price * 0.25"
       WHEN 'christmas eve': "price - 5 - price * 0.35"
    END"
```

so that `price_after_sale` would be 75. I might have time to work on this but would like to know the preferred interface first and if you're interested in having this.

**Comment by rubysolo** [comment]
That sounds like it could be useful.  I think I would prefer the syntax to be:

```
CASE sale
  WHEN 'black friday' THEN price - price * 0.25
  WHEN 'christmas eve' THEN price - 5 - price * 0.35
END
```

Let me know if you get stuck!

**Comment by muescha** [comment]
don't forget a DEFAULT/ELSE keyword

**Comment by rubysolo** [comment]
Closed by 8d78bef
